### ISSUE TITLE
 Scala source code to javadoc

### ISSUE THREAD

**Issue by GeTOUO** [random-maven/scalor-maven-plugin#27]
Do you support compiling and exporting from Scala source code to javadoc?

**Comment by SethTisue** [comment]
sounds like you're looking for https://github.com/lightbend/genjavadoc

**Comment by GeTOUO** [comment]
Thank you for your suggestion. In fact, I encountered some problems using genjavadoc.

My project was built by maven. I called Javadoc during compile phase and inserted other subtasks in Javadoc (generating json-formatted descriptions through doc content). Ultimately, I'll use these JSON data in my restapi documentation.

My problem is:

The Java class generated by genjavadoc can't be compiled, which will cause my build to fail.

**Comment by SethTisue** [comment]
> The Java class generated by genjavadoc can't be compiled, which will cause my build to fail

you think there's a bug in genjavadoc, then? (is there a ticket on that?)

if you're looking for a substitute for genjavadoc, I highly doubt anything else provides similar functionality, I can't be certain nothing exists, but I don't know why someone would build it, given that genjavadoc exists

**Comment by SethTisue** [comment]
oh, I see we were already discussing this at https://github.com/lightbend/genjavadoc/issues/181

**Comment by GeTOUO** [comment]
Yes. When I found genjavadoc, I was very happy. It helped my Scala code go through the Javadoc stage and let me extract the data I needed, but after using it, I found that I could not. Because I need to use genjavadoc translated Java classes in the compilation phase to generate new data in the Javadoc phase, but I can't compile it. Then, when I tried to dig deeper, I found that I couldn't move forward and find an effective way because I knew very little about scala plug-ins. I consulted scaladoc and SBT documents, hoping to find an instance abstraction about scaladoc, and then output my JSON in genjavadoc's life cycle. Data, but I still failed.
